committeeoversightapp/management/commands/import_data.py

```diff
@@ -115,98 +109,165 @@ def add_house_hearings(self):
                 subcommittee2 = row['Subcommittee2']
                 category = row['Category1']
 
-                if name:
-                    # get or create hearing
-                    event, created = Event.objects.get_or_create(jurisdiction_id = self.jurisdiction_id,
-                                    name = name,
-                                    start_date = start_date,
-                                    classification = classification)
-
-                    # save committee1 data
-                    if committee1 and not subcommittee1:
-                        self.new_committee(committee1, event, row_count)
-                    elif subcommittee1 == (committee1 + str(0)):
-                        self.new_committee(committee1, event, row_count)
-                    elif committee1:
-                        self.new_committee(subcommittee1, event, row_count)
-
-                    # save committee2 data
-                    if committee2 and not subcommittee2:
-                        self.new_committee(committee2, event, row_count)
-                    elif committee2 and subcommittee2 == (committee2 + str(0)):
-                        self.new_committee(committee2, event, row_count)
-                    elif committee2:
-                        self.new_committee(subcommittee2, event, row_count)
-
-                    if created:
-
-                        # save event source
-                        source = EventSource.objects.create(event=event, note="spreadsheet", url=source)
-
-                        # save category data
-                        try:
-                            hearing_category = HearingCategory.objects.create(event=event, category_id=category)
-                        except IntegrityError:
-                            self.bad_rows.append("Row " + str(row_count) + ": Unrecognized category " + category + " on " + event.name)
+                # get committees into a edited list format
+                # committee codes with a zero appended indicate "full committee"
+                # and only the full committee will be recorded as an event participant
+                # hearings with a subcommittee listed will only have the subcommittee saved,
+                # as the full committee is attached as a parent
+                committees_filtered = []
```

Comment:
this should be factored out into a method. maybe something like `_participating_committees` and it can take the row dict.

Reply:
(where *this* is lines 117-127)

Reply:
Factored!